### ISSUE TITLE
[FEATURE] align argument usage across different llm providers

### ISSUE THREAD

**Issue by davidberenstein1957** [argilla-io/distilabel#246]
**Is your feature request related to a problem? Please describe.**
Across different LLM providers we seem to be using different ways of configuration.
- We swap the order of arguments.: sometimes `model` is the 1st argument and other times `task`, additionally generation arguments are ordered differently
- We have different naming: sometimes we use `model` sometimes `vllm`, sometimes `openai_api_key` sometimes `api_key`

**Describe the solution you'd like**
I would like to align the naming to `model` and `api_key`. Additionally, I would like to see `client` and `api_key` as last arguments because they are non-default and only apply to certain LLM integrations

**Describe alternatives you've considered**
Using the mixed config we have now.

**Additional context**
N.A.

**Comment by alvarobartt** [comment]
Hi @davidberenstein1957 the main reason behind this is that `task` is always needed and mandatory while e.g. `model` is not mandatory in all the cases as we already provide a default one. Additionally, w.r.t. the rest of the args within the `LLM` those are specific to each LLM engine in most of the scenarios, so would be hard to align, and even in some cases, nonsense. Also providing `generation_kwargs` as a mapping hides visibility from those LLM-specific args, we had that conversation initially when developing it, but at the end we decided to go with `kwargs` even those were differing from one LLM engine to another.

**Comment by davidberenstein1957** [comment]
@alvarobartt I agree. For me it was not clear this was the reasoning behind it. Then we should also change the communication here: https://distilabel.argilla.io/latest/technical-reference/llms/#llm and the various `__rich_repr__`-methods. Besides that, the ordering issue w.r.t. variable introduction persists IMO.
